test/ArcxAnalyticsSdk.test.ts

```diff
@@ -89,290 +110,460 @@ describe('(unit) ArcxAnalyticsSdk', () => {
 
-      expect(eventStub.calledOnceWith('TEST_EVENT', pageAttributes)).to.be.true
-    })
-  })
+        analyticsSdk['sdkConfig'].trackPages = false
+      })
 
-  describe('Automatic Metamask event reporting', () => {
-    let ethereum: MockEthereum
-    let requestStub: sinon.SinonStub
-    let analyticsSdk: ArcxAnalyticsSdk
+      it('emits a FIRST_PAGE_VISIT event with referrer if trackReferrer is true', () => {
+        analyticsSdk['sdkConfig'].trackReferrer = true
 
-    beforeEach(async () => {
-      requestStub = sinon.stub()
-      requestStub.withArgs({ method: 'eth_accounts' }).resolves([TEST_ADDRESS])
-      requestStub.withArgs({ method: 'eth_chainId' }).resolves('0x1')
+        analyticsSdk['_trackFirstPageVisit']()
+        expect(eventStub).calledOnceWithExactly(FIRST_PAGE_VISIT, { referrer: TEST_REFERRER })
 
-      sinon.stub(postRequestModule, 'postRequest').resolves(TEST_IDENTITY)
+        analyticsSdk['sdkConfig'].trackReferrer = false
+      })
 
-      ethereum = new MockEthereum()
-      ethereum.request = requestStub
-      ;(window as any).ethereum = ethereum
+      it('emits a FIRST_PAGE_VISIT event with utm tags if trackUTM is true', () => {
+        analyticsSdk['sdkConfig'].trackUTM = true
 
-      const connectWalletStub = sinon.stub(ArcxAnalyticsSdk.prototype, 'connectWallet')
-      const currentChainIdStub = sinon
-        .stub(ArcxAnalyticsSdk.prototype, <any>'_getCurrentChainId')
-        .resolves(TEST_CHAIN_ID)
-      analyticsSdk = await ArcxAnalyticsSdk.init(TEST_API_KEY, {
-        trackWalletConnections: true,
+        analyticsSdk['_trackFirstPageVisit']()
+        expect(eventStub).calledOnceWithExactly(FIRST_PAGE_VISIT, {
+          utm: {
+            source: TEST_UTM_SOURCE,
+            medium: TEST_UTM_MEDIUM,
+            campaign: TEST_UTM_CAMPAIGN,
+          },
+        })
+
+        analyticsSdk['sdkConfig'].trackUTM = false
       })
-      currentChainIdStub.restore()
-      connectWalletStub.restore()
 
-      sinon.resetHistory()
+      it('emits a FIRST_PAGE_VISIT event with url, referrer and UTM attributes if trackPages, referrer and UTM configs are set to true', () => {
+        analyticsSdk['sdkConfig'].trackPages = true
+        analyticsSdk['sdkConfig'].trackReferrer = true
+        analyticsSdk['sdkConfig'].trackUTM = true
+
+        analyticsSdk['_trackFirstPageVisit']()
+        expect(eventStub).calledOnceWithExactly(FIRST_PAGE_VISIT, {
+          url: TEST_JSDOM_URL,
+          referrer: TEST_REFERRER,
+          utm: {
+            source: TEST_UTM_SOURCE,
+            medium: TEST_UTM_MEDIUM,
+            campaign: TEST_UTM_CAMPAIGN,
+          },
+        })
+
+        analyticsSdk['sdkConfig'].trackPages = false
+        analyticsSdk['sdkConfig'].trackReferrer = false
+        analyticsSdk['sdkConfig'].trackUTM = false
+      })
     })
 
-    afterEach(sinon.restore)
+    describe('#_trackPagesChange', () => {
+      it('registers an on-click event listener', () => {
```

Comment:
`_trackPagesChange: registers an on-click event listener` 
I don't get the idea to create context for single test

Reply:
as described above